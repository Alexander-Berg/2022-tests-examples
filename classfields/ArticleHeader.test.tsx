import React from 'react';
import { shallow } from 'enzyme';

import contextMock from 'autoru-frontend/mocks/contextMock';
import mockStore from 'autoru-frontend/mocks/mockStore';

import articleMock from 'auto-core/react/dataDomain/mag/articleMock';
import configMock from 'auto-core/react/dataDomain/config/mock';

import type { ArticleHeaderProps } from './ArticleHeader';
import ArticleHeader from './ArticleHeader';

const initialArticle = articleMock.value();

function shallowRenderComponent(props: ArticleHeaderProps) {
    return shallow(
        <ArticleHeader { ...props }/>,
        {
            context: {
                ...contextMock,
                store: mockStore({
                    config: configMock.value(),
                }),
            },
        },
    );
}

it('должен отрендерить вариант с предыдущей статьёй', () => {
    const wrapper = shallowRenderComponent({
        title: initialArticle.title,
        withLinkComments: true,
        relatedArticle: initialArticle.before,
    });

    expect(wrapper.find('.ArticleHeader__previousArticle')
        .prop('href'))
        .toContain(`linkMag/mag-article/?article_id=${ initialArticle.before?.urlPart }`);
});

it('должен отрендерить обычный вариант с блоком "Поделиться"', () => {
    const wrapper = shallowRenderComponent({
        title: initialArticle.title,
    });

    expect(wrapper.find('Memo(Share)')).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
});
