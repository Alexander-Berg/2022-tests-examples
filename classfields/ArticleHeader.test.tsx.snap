// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен отрендерить обычный вариант с блоком "Поделиться" 1`] = `
<header
  className="ArticleHeader"
>
  <ForwardRef(SectionBlock)
    className="ArticleHeader__content"
    gap="none"
    size="l"
  >
    <div
      className="ArticleHeader__autoruLogo"
    >
      <a
        className="MagHeaderLogo__link"
        href="link/index/?"
      >
        Авто.ру
      </a>
    </div>
    <div
      className="ArticleHeader__magLogo"
    >
      <a
        className="MagHeaderLogo__link"
        href="linkMag/mag-index/?"
      >
        Журнал Авто.ру
      </a>
    </div>
    <Memo(Share)
      className="ArticleHeader__share"
      iconSize="20"
      title="Тестовая статья со всеми блоками для тестирования (Не редактируйте эту статью)"
    />
  </ForwardRef(SectionBlock)>
</header>
`;
